### ISSUE TITLE
Error with `engines` set to asterisk

### ISSUE THREAD

**Issue by brettz9** [mysticatea/eslint-plugin-node#155]
With the following config:

```js
module.exports = {
  "extends": ["plugin:node/recommended"],
  parserOptions: {
    sourceType: "module"
  },
  "plugins": ["node"],
  "rules": {
    "node/no-unsupported-features/es-builtins": ["error"]
  }
};
```

and with this in `package.json`:

```json
"engines": {
    "node": "*"
}
````

I get this error:

> TypeError: Error while loading rule 'node/no-unsupported-features/es-syntax': Invalid Version: [object Object]
> Occurred while linting /Users/brett/handle-node/rollup.config.js

Setting it to `>= 0.1.14` or such works fine.

The asterisk is allowable here per https://docs.npmjs.com/files/package.json#engines , and I would think this value would still make since with this rule, e.g., to support any past Node version or just to debug and see which features one is using that are not supported.

**Comment by mysticatea** [comment]
Thank you for the report.

Looks like a bug in `semver` package: https://github.com/npm/node-semver/issues/266
I'm waiting for fixed.

**Comment by brettz9** [comment]
Should no longer be blocking as of https://github.com/npm/node-semver/tree/v6.1.0

**Comment by mysticatea** [comment]
Thank you for the notice!

So it's available now because it's a minor version up. (this depends on `semver@^6.0.0`)
